packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -33,7 +33,7 @@ export default function buildConfigChain(opts: Object = {}) {
   builder.mergeConfig({
     options: opts,
     alias: "base",
-    dirname: filename && path.dirname(filename),
```

Comment:
True! I'd probably remove that from there so this declaration is clear up front here.